pkg/controller/jobframework/reconciler.go

```diff
@@ -409,6 +474,11 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 
 // startJob will unsuspend the job, and also inject the node affinity.
 func (r *JobReconciler) startJob(ctx context.Context, job GenericJob, object client.Object, wl *kueue.Workload) error {
+	err := r.setFinalizer(ctx, wl, true)
```

Comment:
This will require an extra API request. Why not add the finalizer during workload creation in ConstructWorkload?